### PR TITLE
Allow query parameters to be passed to custom changes filters

### LINKED ISSUE #402: Cannot set filter parameters in changes() call
The Feed._validate method makes it impossible to pass arbitrary query parameters to a _changes filter design document when calling CouchDatabase.changes().
https://github.com/cloudant/python-cloudant/blob/4435d84f24dd3563e4abd2929e2953eeba0d5b89/src/cloudant/feed.py#L114
If the key is not in arg_types, it should assume it is a custom parameter directed at the filter and hope. Maybe just check if it is an int, float, or str, as these are the only supported types for the query.

I suppose the same holds for Cloudant._db_updates().

### PR DESCRIPTION
**Pull request by smithsz**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Allow arbitrary query parameters to be passed to custom changes filters.

Fixes #402.

## Approach

Add new `ANY_ARG` and `ANY_TYPE` to list of valid changes arguments:

```py
_CHANGES_ARG_TYPES = {
    'conflicts': (bool,),
    'doc_ids': (list,),
    'filter': (STRTYPE,),
    'include_docs': (bool,),
    'style': (STRTYPE,),
    ANY_ARG: ANY_TYPE  # pass arbitrary query parameters to a custom filter
}
```

## Schema & API Changes

No change.

## Security and Privacy

No change.

## Testing

Includes additional unit test.

## Monitoring and Logging

No change.

### CODE REVIEW

File: tests/unit/changes_tests.py [hunk 1]
```diff
@@ -465,14 +465,29 @@ def test_get_feed_using_doc_ids(self):
...
+            filter='mailbox/new_mail',
+            foo='bar',  # query parameters to a custom filter
+            include_docs=False
+        )
+        params = feed._translate(feed._options)
+        self.assertEquals(params['filter'], 'mailbox/new_mail')
+        self.assertEquals(params['foo'], 'bar')
+        self.assertEquals(params['include_docs'], 'false')
+
+    # def test_invalid_argument(self):
```
Comment (smithsz) [inline tests/unit/changes_tests.py]: This test is no longer valid as passing `foo='bar'` is now accepted under `ANY_ARG: ANY_TYPE`.

File: tests/unit/changes_tests.py [hunk 2]
```diff
@@ -465,14 +465,20 @@ def test_get_feed_using_doc_ids(self):
         self.assertSetEqual(set([x['id'] for x in changes]), expected)
         self.assertTrue(str(feed.last_seq).startswith('100'))
 
-    def test_invalid_argument(self):
```
Comment (smithsz) [inline tests/unit/changes_tests.py]: This test is no longer valid as passing foo='bar' is now accepted under ANY_ARG: ANY_TYPE.

File: src/cloudant/feed.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
```
Comment (ricellis) [inline src/cloudant/feed.py]: This needs a copyright update

**Comment by emlaver** [comment]
@smithsz There's a couple cases that cause a CloudantArgumentError `___ type is not callable`:
```
feed = Feed(self.db, foo={}) # type is dict
feed = Feed(self.db, foo=object_var) # type is object
```
We should probably only allow `int`, `float`, `string`, and `bool` types.

**Comment by smithsz** [comment]
I think we should serialise `foo={}`. Obviously serialising `foo=object()` will never work and should throw.

How about making `json.dumps` the default type converter? (as per fb490ca712c017789358232d1118f8e517011120)

**Pull request merged**